src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,67 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    private int randomeRadius = 0;
+    private int squareRandomeSide = 0;
+    private int rectangleRandomASide = 0;
+    private int rectangleRandomBSide = 0;
+    private int firstRandomeLeg = 0;
+    private int secondRandomeLeg = 0;
+    private int trapezoidRandomeASide = 0;
+    private int trapezoidRandomeBSide = 0;
+    private int trapezoidRandomeHeigth = 0;
+
+    public Figure getRandomFigure() {
+        String figureType = getRandomeType();
+        Figure figure = null;
+        switch (figureType) {
+            case "circle":
+                randomeRadius = random.nextInt(100);
+                figure = new Circle(randomeRadius, colorSupplier.getRandomColor());
+                break;
```

Comment:
Move it and all other `case` blocks statements into the separate methods, which returns the random figure.